### ISSUE TITLE
Does Dragging blur platform and author considerations?

### ISSUE THREAD

**Issue by mbgower** [w3c/wcag2ict#6]
In https://github.com/w3c/wcag/issues/1307 the user asks whether iOS accommodation features (specifically assistive touch) are sufficient to support users who lack the ability to easily invoke a drag action.

Part of the AGWG's official response is that such accommodations:
>  are not universally available and would not qualify as Sufficient Technique for a wider accessibility baseline covering different platforms and user agents. 

I think this response poses an interesting scenario we need to consider, and also points to a potential logic puzzle in the development of this SC (and arguably in Pointer Gestures as well).

First, we say that Assistive Touch is not an acceptable support because it is not universal. What if it was/is?

I can see the guidance getting in an odd situation where every touch-OS provides affordances for simple-touch drag and drop, and yet we continue to insist author's solve this. The analogy would be making authors responsible for key repetition and other keyboard operation simplifiers even though they are offered at the OS level (and piggy-backed on by any web author).

This new SC grew from a desire to further the guidance introduced in 2.1 for Pointer Gestures, which itself was the result of the Mobile Accessibility Task Force trying to ensure touch was made more accessible.

I understand the value of encouraging authors to offer simple, single pointer actions. But does this SC become redundant when hardware providers are required to provide single-touch mechanisms (through other standards)? Is that already the case with Android, iOS and Windows 10 systems? When are we placing undue burden on an author for a common interaction already solved by an elegant accessibility setting at the OS level? And when will we know this is reality?

Second, I think there is a related logic puzzle to solve. In this hypothetical, we have a user who cannot do complex gestures but wants to use a touch device. How does the user ever get to the web page where we are insisting simple touch interactions are required? They obviously need to first open and operate their device to even reach a web page. It seems to me there are a few possible answers:

1. The device interaction is designed so that there is no reliance on gestures and dragging for all its functions. (There is a method of achieving all functions with simple operations, similar to the WCAG SCs).
1. The device has an accommodation that allows a user to invoke all required complex functions with simple operations (The device offers something like Assistive Touch)
1. The user figures out workarounds for interactions that lack accommodations, which allows them to operate the device to a degree.
1. The user does not use the device.

In scenario 1, if the system itself supports simple pointer interactions for all functions, then the need to have SCs that require the same on author-controlled web pages seems critical. 
**_However_**, it is easy to demonstrate that mobile systems do not meet scenario 1. For instance, the only touch way to scroll inside a mobile touch browser is by flicking or dragging up and down. There is no simple touch equivalent that I am aware of. Therefore, the only ways someone can even _view_ a web page on a mobile touch screen (whose content exceeds the viewport) is to do actions the user in our hypothetical cannot achieve -- or to invoke a different modality, like keyboard operation.

For scenario 2, if the system provides accommodations (accessibility features) that allow someone to complete all functions with simple pointer interactions, then as long as authors invoke those same functions in a way that is supported, there is no need to have the SCs for Pointer Gesture and Dragging in their _current_ state. The user is already able to operate the OS with the accommodations. The SC can be restricted to ensuring authors create content that supports the platform affordances.

For scenario 3, if workarounds exists for some interactions that lack accommodations, the user may or may not be able to reach the web pages, where those workarounds may or may not also work for authoring guidance. The SCs would have value. However, without having requirements for the platform and user agent, the SC requirements are unlikely to solve many of the user issues.

For scenario 4, where devices fail to support touch users with reduced ability, creating requirements for web authors to support touch is not going to matter (for those devices).

I'm not sure we have sufficient research to know what blending of the above scenarios most closely reflects reality, nor what the cost/benefit is of trying to have web authors individually solve interaction challenges that are considerations across a touch device's entire capabilities.

**Comment by maryjom** [comment]
Moved to WCAG2ICT repository as the new TF will work to address issues tagged as WCAG2ICT from the WCAG repository.

**Comment by detlevhfischer** [comment]
@mbgower 
> As well, one of your workaround examples ("move them via a dropdown menu") invariably involves dragging on mobile. So the solution to dragging itself involves dragging.

How so? you single tap/click the menu button; menu opens and displays a list of options;  you move the pointer and single tap/click an option? (But maybe I misunderstand...)

**Comment by maryjom** [comment]
@GreggVan This issue was an existing one related to dragging. Can you take a look at this to ensure the draft you made addresses this issue as well? Thanks!

**Comment by maryjom** [comment]
@mbgower Wondering how you think that WCAG2ICT could or should clear this up?  We don't provide techniques or requirements. Please check the current WCAG2ICT guidance on [Dragging movements](https://w3c.github.io/wcag2ict/#applying-sc-2-5-7-dragging-movements-to-non-web-documents-and-software). Does this section sufficient interpretation and information, or do you feel something else is needed?

**Comment by mbgower** [comment]
You have retained Notes 1 and 3
> This requirement applies to [content](https://w3c.github.io/wcag2ict/#content-on-and-off-the-web) that interprets pointer actions (i.e. this does not apply to actions that are required to operate the user agent or assistive technology).
>This requirement applies to [[software](https://w3c.github.io/wcag2ict/#software)] that interprets pointer actions (i.e. this does not apply to actions that are required to operate the [underlying platform software] or assistive technology).

What if the software _is_ a user agent, AT or platform software? Shouldn't these notes include some qualifiers for this kind of software?

**Comment by maryjom** [comment]
@mbgower Note 1 is scoped specifically to non-web documents. User agents are non-web software. 

Note 3 is scoped to non-web software. For that, the note is specifically about software applications that have some user agent or platform software underneath - to indicate the software applications aren't responsible for fixing issues with any dragging required by their underlying OS or User Agent for standard UI elements supplied/supported by the underlying software. 

Perhaps another note is needed to talk about user agents and OS, or we need to figure out how to show that a particular software layer (application, user agent, OS) are only responsible for dragging movements that that implemented/interpreted by that particular layer of software. What do you think?

**Comment by maryjom** [comment]
@mbgower The adjustments to the WCAG2ICT guidance are being developed using Issue #284. Once that is settled, I think this issue can be closed. Agree?

**Comment by maryjom** [comment]
WCAG2ICT TF answer:

Mike, 
You raised a good point regarding the need for a separation of responsibilities between the author, user agent, and platform software. Glad you re-raised and clarified the issue in your [review of 2.5.7 Dragging Movements](https://github.com/w3c/wcag2ict/issues/268#issuecomment-1839873691). The WCAG2ICT Task Force has developed guidance for non-web contexts in the notes to address the concern. This same guidance is provided for other success criteria that also interpret pointer actions, namely 2.5.1 Pointer Gestures and 2.5.2 Pointer Cancellation. The notes read:

NOTE: This requirement applies to **[user agents and other software applications that interpret]** pointer actions (i.e. this does not apply to actions that are required to operate the **[underlying platform software]** or assistive technology).

NOTE: This requirement also applies to **[platform software]** that interprets pointer actions. This does not apply to actions that are required to operate the assistive technology.

See PR #296 for the changes. The WCAG2ICT Task Force approved the above notes in our 11 January meeting. The PR incorporates those updates into the document.

Related issue: #284 (opened due to the AG WG review of 2.5.7) where the Task Force developed these notes.

@mbgower Please close this issue if your concerns have been satisfied. Thanks!

**Comment by maryjom** [comment]
@ mbgower I am closing this issue, as you reviewed the changes already based on the issue and PR. If there's anything further, please open a new issue.
